File: hyperLocal-multivendor-eCommerce-backend-admin-seller-v1.0.0/app/Models/Address.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * @method static where(array $array)
 */
class Address extends Model
{
    protected $fillable = [
        'user_id',
        'address_line1',
        'address_line2',
        'city',
        'landmark',
        'state',
        'zipcode',
        'mobile',
        'address_type',
        'country',
        'country_code',
        'latitude',
        'longitude',
    ];

    /**
     * The attributes that should be cast.
     */
    protected $casts = [
        'id' => 'integer',
        'user_id' => 'integer',
        'address_line1' => 'string',
        'address_line2' => 'string',
        'city' => 'string',
        'landmark' => 'string',
        'state' => 'string',
        'zipcode' => 'string',
        'mobile' => 'string',
        'address_type' => 'string',
        'country' => 'string',
        'country_code' => 'string',
        'latitude' => 'float',
        'longitude' => 'float',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: hyperlocal-dioxus-customer/src/types/address.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Address {
    pub id: Option<i32>,
    #[serde(rename = "address_line1")]
    pub address_line1: Option<String>,
    #[serde(rename = "address_line2")]
    pub address_line2: Option<String>,
    pub city: Option<String>,
    pub state: Option<String>,
    pub landmark: Option<String>,
    #[serde(rename = "postal_code")]
    pub postal_code: Option<String>,
    pub latitude: Option<f64>,
    pub longitude: Option<f64>,
    #[serde(rename = "address_type")]
    pub address_type: Option<String>,
    #[serde(rename = "is_default")]
    pub is_default: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AddressRequest {
    #[serde(rename = "address_line1")]
    pub address_line1: String,
    #[serde(rename = "address_line2")]
    pub address_line2: Option<String>,
    pub city: String,
    pub state: String,
    pub landmark: Option<String>,
    #[serde(rename = "postal_code")]
    pub postal_code: Option<String>,
    pub latitude: f64,
    pub longitude: f64,
    #[serde(rename = "address_type")]
    pub address_type: String,
    #[serde(rename = "is_default")]
    pub is_default: Option<bool>,
}
